artemis-core-client/src/main/java/org/apache/activemq/artemis/api/core/Message.java

```diff
@@ -726,20 +728,44 @@ default long getWholeMessageSize() {
 
    /**
     * @return Returns the message properties in Map form, useful when encoding to JSON
+    * @param valueSizeLimit
     */
-   default Map<String, Object> toPropertyMap() {
+   default Map<String, Object> toPropertyMap(int valueSizeLimit) {
       Map map = new HashMap<>();
       for (SimpleString name : getPropertyNames()) {
          Object value = getObjectProperty(name.toString());
          //some property is SimpleString, which is not available for management console
          if (value instanceof SimpleString) {
             value = value.toString();
          }
+         value = truncate(value, valueSizeLimit);
          map.put(name.toString(), value);
       }
       return map;
    }
 
+   static Object truncate(final Object value, final int valueSizeLimit) {
+      Object result = value;
+      if (valueSizeLimit > 0) {
```

Comment:
I would expect truncate 0 to give me zero, normally expect to not truncate or unlimited to be denoted by -1

Reply:
ok, -1 it will be.